ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/Leaf.java

```diff
@@ -74,48 +73,49 @@ public String getMessage() {
      *
      * It is built directly from a RaoData on which a systematic sensitivity analysis could hav already been run or not.
      */
-    // This constructor is useful to mock SystematicSensitivityComputation in LeafTest
-    Leaf(RaoData raoData, RaoParameters raoParameters, SystematicSensitivityComputation systematicSensitivityComputation, IteratingLinearOptimizer iteratingLinearOptimizer) {
+    // This constructor is useful to mock SystematicSensitivityInterface in LeafTest
+    Leaf(RaoData raoData, RaoParameters raoParameters) {
         this.networkActions = new HashSet<>(); // Root leaf has no network action
         this.raoParameters = raoParameters;
-        this.systematicSensitivityComputation = systematicSensitivityComputation;
-        this.iteratingLinearOptimizer = iteratingLinearOptimizer;
         this.raoData = raoData;
         initialVariantId = raoData.getInitialVariantId();
+        //TODO: check in parameters if we need PTDFs for loopflows or not
+        systematicSensitivityInterface = SystematicSensitivityInterface
+            .builder()
+            .withDefaultParameters(raoParameters.getDefaultSensitivityComputationParameters())
+            .withFallbackParameters(raoParameters.getFallbackSensitivityComputationParameters())
+            .withRangeActionSensitivities(raoData.getAvailableRangeActions(), raoData.getCnecs())
+            .build();
+
```

Comment:
we could maybe put this in a dedicated method to avoid the duplication of the todo

Reply:
initialization is now done in a dedicated method (the variable is no longer final, which may end up being a good thing if we wish to modify the interface, for instance in the scenario where we wish to compute the PTDFs only once at the beginning)